tmt/steps/report/reportportal.py

```diff
@@ -0,0 +1,150 @@
+
+
+def import_junit_xml() -> None:
+    """
+    Import junit_xml module only when needed
+
+    Until we have a separate package for each plugin.
+    """
+    global junit_xml
+    try:
+        import junit_xml
+    except ImportError:
+        raise tmt.utils.ReportError(
+            "Missing 'junit-xml', fixable by 'pip install tmt[report-junit]'.")
+
+
+@dataclasses.dataclass
+class ReportReportPortalData(tmt.steps.report.ReportStepData):
+    url: Optional[str] = tmt.utils.field(
+        option="--url",
+        metavar="URL",
+        default=os.environ.get("TMT_REPORT_REPORTPORTAL_URL"),
+        help="The URL of the ReportPortal instance where the data should be sent to.")
+    token: Optional[str] = tmt.utils.field(
+        option="--token",
+        metavar="TOKEN",
+        default=os.environ.get("TMT_REPORT_REPORTPORTAL_TOKEN"),
+        help="The token to use for upload to the ReportPortal instance.")
+    project: Optional[str] = tmt.utils.field(
+        option="--project",
+        metavar="PROJECT",
+        default=None,
+        help="The project name which is used to create the full URL.")
+    launch_name: Optional[str] = tmt.utils.field(
+        option="--launch-name",
+        metavar="NAME",
+        default=None,
+        help="The launch name (base name of run id used by default).")
+
+
+@tmt.steps.provides_method("reportportal")
+class ReportReportPortal(tmt.steps.report.ReportPlugin):
+    """
+    Report test results to a ReportPortal instance
+
+    Requires a TOKEN for authentication, a URL of the ReportPortal
+    instance and the PROJECT name. In addition to command line options
+    it's possible to use environment variables to set the url and token:
+
+        export TMT_REPORT_REPORTPORTAL_URL=...
+        export TMT_REPORT_REPORTPORTAL_TOKEN=...
+
+    The optional launch NAME is passed to ReportPortal. Assuming the URL
+    and TOKEN variables are provided by the environment, the config can
+    look like this:
+
+        report:
+            how: reportportal
+            project: baseosqe
+            launch-name: maven
+    """
+
+    _data_class = ReportReportPortalData
+
+    def go(self) -> None:
+        """
+        Read executed tests, prepare junit, compress it to a zip zile and
+        send it to the ReportPortal instance.
+        """
+
+        super().go()
+
+        # Check the data, show interesting info to the user
+        # Required fields are: url, token and project
+        server = self.get("url")
+        if not server:
+            raise tmt.utils.ReportError("No ReportPortal server url provided.")
+        server = server.rstrip("/")
+
+        token = self.get("token")
+        if not token:
+            raise tmt.utils.ReportError("No ReportPortal token provided.")
+
+        project = self.get("project")
+        self.info("project", project, color="green")
+        if not project:
+            raise tmt.utils.ReportError("No ReportPortal project provided.")
```

Comment:
I'd maybe switch the order here: if the exception is raised, it's clear what the value of `project` was, and it'd not result in a weird `project: None` followed by an exception.

Reply:
Good point!

Reply:
Fixed in 8b4fd34e.